java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/LspAnalyser.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.lsp.server.protocol;
+
+import org.netbeans.api.annotations.common.NonNull;
+import org.netbeans.modules.java.lsp.server.ui.AnalyserConfig;
+import org.netbeans.modules.java.source.usages.BinaryAnalyser;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ *
+ * @author Dusan Balek
+ */
+@ServiceProvider(service = BinaryAnalyser.Config.class, position = 1000)
```

Comment:
Q: should be this service active (registered) during regular IDE operation ? If not, the `@ServiceProvider` should move to `nbcode`.

Reply:
This service is intended to be active (registered) during both regular IDE and LSP operation. It replaces the default `BinaryAnalyser.Config` implementation. Its functionality can be adjusted form `nbcode` using `LSPAnalyserConfig` without implementation dependency of `nbcode` suite on `Java Source Base`.